DashWallet/Sources/UI/DashPay/Contacts/GlobalSearch/Model/DWUserSearchModel.m

```diff
@@ -165,42 +166,37 @@ - (void)performSearchWithQuery:(NSString *)query offset:(uint32_t)offset {
-                              strongSelf.searchRequest.items = items;
-                              [strongSelf.delegate userSearchModel:strongSelf completedWithItems:items];
-                          }
-                          else {
-                              strongSelf.searchRequest.hasNextPage = NO;
-                              [strongSelf.delegate userSearchModel:strongSelf completedWithError:errors.firstObject];
-                          }
-                      }];
+    self.request = [manager searchIdentitiesByNamePrefix:query
+                                                inDomain:@"dash"
+                                              startAfter:self.searchRequest.lastItem
+                                                   limit:LIMIT
+                                          withCompletion:^(BOOL success, NSArray<DSBlockchainIdentity *> *_Nullable blockchainIdentities, NSArray<NSError *> *_Nonnull errors) {
+                                              __strong typeof(weakSelf) strongSelf = weakSelf;
+                                              if (!strongSelf) {
+                                                  return;
+                                              }
+                                              NSAssert([NSThread isMainThread], @"Main thread is assumed here");
+                                              // search query was changed before results arrive, ignore results
+                                              if (!strongSelf.searchRequest || ![strongSelf.searchRequest.trimmedQuery isEqualToString:query]) {
+                                                  return;
+                                              }
+                                              strongSelf.searchRequest.requestInProgress = NO;
+                                              if (success) {
+                                                  NSMutableArray<id<DWDPBasicUserItem, DWDPBlockchainIdentityBackedItem>> *items = strongSelf.searchRequest.items ? [strongSelf.searchRequest.items mutableCopy] : [NSMutableArray array];
+                                                  for (DSBlockchainIdentity *blockchainIdentity in blockchainIdentities) {
+                                                      id<DWDPBasicUserItem, DWDPBlockchainIdentityBackedItem> item = [strongSelf.itemsFactory itemForBlockchainIdentity:blockchainIdentity];
+                                                      [items addObject:item];
+                                                  }
+                                                  strongSelf.searchRequest.hasNextPage = blockchainIdentities.count >= LIMIT;
+                                                  strongSelf.searchRequest.items = items;
+                                                  strongSelf.searchRequest.lastItem = uint256_data([[[items lastObject] blockchainIdentity] uniqueID]);
```

Comment:
You can just use uniqueIDData I think instead.

Reply:
+